### ISSUE TITLE
Lunar Diagnostics Report [A8D0AB78C2363803FB94D226F43D8F4284A5C1ABB867E2380A1768204634ABAF] [3.2.3]

### ISSUE THREAD

**Issue by eboddington** [alin23/Lunar#165]
# Issue details
- Was the diagnostics process able to change the brightness on your external monitor(s)?
    No

- Mac device where Lunar is installed (Macbook Pro 2019, iMac, Mac Mini, Hackintosh etc.):
    15" MacBook Pro 2017 - AMD Radeon Pro 560 4 GB

- Monitor connection to the Mac device (HDMI-to-USB-C, USB-C-to-USB-C, miniDisplayPort-to-DisplayPort etc.):
    USB-C/TB3 to DisplayPort (1.2) - 2560x1440 @ 144Hz
    Previously I'd tried the Apple "USB-C Digital AV Multiport Adapter" and HDMI - 2560x1440 @ 60Hz, which also did not work.

- Using an USB Docking Station or Hub:
    No (though I had tried doing so previously, see above, and it also didn't work)

- Lunar mode used (check it in the top-right corner of the Lunar interface)
    I tried all of them, the monitor brightness never changed.

- (only if you know how to compile a C program) Does this utility work for you? https://github.com/kfix/ddcctl
    Compiled and ran from master, it was able to identify my screen and correctly read the current brightness, but unable to change it.

# Issue description:
No brightness or contrast control worked.

**Comment by alin23** [comment]
> Compiled and ran from master, it was able to identify my screen and correctly read the current brightness, but unable to change it.

That's very odd, I never encountered a case where reading works but writing doesn't. 99% of the time it's the other way around.
Does this happen with both `make amd` and `make intel` in ddcctl?

https://github.com/kfix/ddcctl#install

**Comment by eboddington** [comment]
> That's very odd, I never encountered a case where reading works but writing doesn't. 99% of the time it's the other way around.
> Does this happen with both `make amd` and `make intel` in ddcctl?
> 
> https://github.com/kfix/ddcctl#install

I was previously using `make amd`, since I thought external displays are always connected to the dedicated graphics, but I've just tried with `make intel` too, and it's the same. I've tried changing the brightness using the monitor's controls, and confirmed that `ddcctl` correctly reads the new value, but it's still unable to change it:
```
# ~/Downloads/ddcctl on git:master o [11:51:53] 
$ ./ddcctl -d 1 -b \?
D: NSScreen #478272515 (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling display 1's EDID
I: got edid.serial: #<redacted>
I: got edid.name: XB271HU
D: action: b: ?
D: querying VCP control: #16 =?
I: VCP control #16 (0x10) = current: 22, max: 100

# ~/Downloads/ddcctl on git:master o [11:51:56] 
$ ./ddcctl -d 1 -b 70
D: NSScreen #478272515 (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling display 1's EDID
I: got edid.serial: #<redacted>
I: got edid.name: XB271HU
D: action: b: 70
D: setting VCP control #16 => 70

# ~/Downloads/ddcctl on git:master o [11:52:05] 
$ ./ddcctl -d 1 -b \?
D: NSScreen #478272515 (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling display 1's EDID
I: got edid.serial: #<redacted>
I: got edid.name: XB271HU
D: action: b: ?
D: querying VCP control: #16 =?
I: VCP control #16 (0x10) = current: 22, max: 100
```

**Comment by alin23** [comment]
That's unfortunate.. Lunar uses most of the same code as ddcctl for the DDC implementation so if ddcctl doesn't work, we can't do much to fix that in Lunar.

Maybe the monitor has a setting that prevents DDC writes. Try going through the monitor settings and look for fancy features like dynamic/adaptive brightness/contrast or for settings that contain words like _lock_

**Comment by eboddington** [comment]
I went through all of the settings and found nothing which seemed like it would affect this 😞 

There's an adaptive contrast setting, which is set to 0. I've found [this](https://community.acer.com/en/discussion/576132/xb271hu-brightness-control-over-ddc-ci-protocol) page where someone else found the same thing - I tried what they suggest, turning it on in factory mode, and Lunar can indeed change the brightness then. But this is a sort of debug mode with an overlay so I can't use it in this mode normally.

It seems like the monitor simply has it locked ordinarily.
